purchase_triple_discount/models/account_invoice.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2017 Tecnativa - David Vidal
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    def _prepare_invoice_line_from_po_line(self, line):
+        vals = super(AccountInvoice,
+                     self)._prepare_invoice_line_from_po_line(line)
+        vals['discount2'] = line.discount2
```

Comment:
@chienandalu I think, It's batter :smile: 
```
return {
    'discount2': line.discount2,
    'discount3': line.discount3
}
```

Reply:
No, what you propose is wrong. You're ignoring the values from the super()

Reply:
Typo Misteks

val.update( {
    'discount2': line.discount2,
    'discount3': line.discount3
})
return val

Reply:
Why it's better? You are writing 4 lines of code against 2 and no performance difference.